ratpack-bearer-auth/src/main/java/st/ratpack/auth/CachingTokenValidator.java

```diff
@@ -4,44 +4,88 @@
 import com.github.benmanes.caffeine.cache.LoadingCache;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import ratpack.exec.ExecController;
 import ratpack.exec.Promise;
 
 import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
 
 public class CachingTokenValidator implements TokenValidator {
 
 	private final TokenValidator upstreamValidator;
 	private final LoadingCache<String, Promise<Optional<OAuthToken>>> cache;
+	private final ExecController execController;
 	private static Logger logger = LoggerFactory.getLogger(CachingTokenValidator.class);
 
-	public CachingTokenValidator(TokenValidator upstreamValidator) {
+	public CachingTokenValidator(TokenValidator upstreamValidator,
+	                             long timoutMinutes, TimeUnit timeUnit,
+	                             boolean refresh,
+	                             ExecController execController) {
 		this.upstreamValidator = upstreamValidator;
+		this.execController = execController;
 
-		cache = Caffeine.<String, Promise<Optional<OAuthToken>>>newBuilder()
-			.maximumSize(10000L)
-			.expireAfterWrite(5L, TimeUnit.MINUTES)
-			//			.executor(Execution.current().getEventLoop())  Don't do this it makes Ratpack hang.
-			.build(this::loadCache);
+		if (refresh) {
```

Comment:
Should we split this into two classes instead seems easier to maintain

Reply:
@beckje01 much of the logic on cache load is shared and there are weird issues with using an abstract class to handle that. You can't reference a method in a cache builder before constructing a super class. I could break the logic out into a static method with all dependencies passed in I suppose